### PR TITLE
refactored CMS start up - it no longer uses falls back on default con…

### PR DESCRIPTION
**Pull request by daiLlew**
- Refactored CMS start up so it no longer falls back on default content if there is an error starting up. 
  
  Previously any error starting up would be caught and it would attempt to start up again using the default content. This functionality is misleading, masks issues and makes the CMS seem healthy when its not. 

  With these new changes any error initialising the CMS will causing the application to shutdown with an explicit message about the error encountered. 
- Updated `Restolino` to latest version.
- Added `@Priority(n)` annotations to init classes to guarantee start up order - Reader 1st, CMS 2nd.

### CODE REVIEW

File: zebedee-cms/src/main/java/com/github/onsdigital/zebedee/api/Root.java [hunk 1]
```diff
@@ -126,15 +85,61 @@ public static void init() {
...
 
+    /**
+     * Validate the zebedee root path meets the requirements - not null/empty, the file exists and is a directory.
+     * Throws {@link RuntimeException} if any of the checks fail.
+     *
+     * @param rootDir the path to validate.
+     * @return the root dir value as a {@link Path} object.
+     */
+    private static Path validateZebedeeRootConfig(String rootDir) {
+        if (StringUtils.isEmpty(rootDir)) {
```
Comment (s-xton) [inline zebedee-cms/src/main/java/com/github/onsdigital/zebedee/api/Root.java:103]: This check does not seem to get called, see 

ReaderConfiguration line 105

File: zebedee-cms/src/main/java/com/github/onsdigital/zebedee/api/Root.java [hunk 2]
```diff
@@ -126,15 +85,61 @@ public static void init() {
...
+
+        if (!Files.isDirectory(p)) {
+            throw new RuntimeException(
+                    format("invalid zebedee_root config:{0} the specified path is not a directory", rootDir));
+        }
+
+        info().data(ZEBEDEE_ROOT, rootDir)
+                .log("successfully validated zebedee_root config proceeding beginning CMS initialization");
+        return p;
+    }
```
Comment (s-xton) [inline zebedee-cms/src/main/java/com/github/onsdigital/zebedee/api/Root.java:121]: just a thought, I was wondering if you check for one or more of the known directories here as Zebedee will start up if the env variable to set to an empty directory ?

**Review by s-xton — CHANGES_REQUESTED** [review]

File: zebedee-cms/src/main/java/com/github/onsdigital/zebedee/api/Root.java (hunk 1 above)
Comment (daiLlew) [reply zebedee-cms/src/main/java/com/github/onsdigital/zebedee/api/Root.java:103]: ReaderConfiguration line 153 assigns it to the class field which is then used in the subsequent constructor logic.

**Review by s-xton — DISMISSED** [review]

**Comment by s-xton** [comment]
Tested with empty zedebee_root
Tested with zedebee_root to invalid location
Tested with zedebee_root to a file location
Tested with zedebee_root to a directory location

commented issue is down to race condition in existing framework

**Review by Marianasilvago — APPROVED** [review]
LGTM

**Review by Marianasilvago — APPROVED** [review]
LGTM

File: zebedee-cms/src/main/java/com/github/onsdigital/zebedee/Init.java [hunk 3]
```diff
@@ -16,11 +17,13 @@
...
 import com.github.onsdigital.zebedee.reader.ZebedeeReader;
 
-import static com.github.onsdigital.logging.v2.event.SimpleEvent.info;
+import static com.github.onsdigital.zebedee.logging.CMSLogEvent.error;
+import static com.github.onsdigital.zebedee.logging.CMSLogEvent.info;
 
 /**
  * Created by bren on 31/07/15.
  */
+@Priority(2)
```
Comment (daiLlew) [inline zebedee-cms/src/main/java/com/github/onsdigital/zebedee/Init.java:26]: Annotation specifies the order the `Startup` instances should run. The app logging is configured within Reader so we want reader to start before CMS so we don't miss any log events.

**Review by franmoore05 — APPROVED** [review]
LGTM

**Pull request merged**